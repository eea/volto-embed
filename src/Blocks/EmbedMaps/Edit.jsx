import React from 'react';
import { compose } from 'redux';
import { injectIntl } from 'react-intl';
import BlockDataForm from '@plone/volto/components/manage/Form/BlockDataForm';
import { SidebarPortal } from '@plone/volto/components';
import View from './View';
import getSchema from './schema';

const Edit = (props) => {
  const { block, data, selected, onChangeBlock } = props;
  const schema = React.useMemo(() => getSchema(props), [props]);

  return (
    <React.Fragment>
      <View {...props} mode="edit" />
      <SidebarPortal selected={selected}>
        <BlockDataForm
          block={block}
          schema={schema}
          title={schema.title}
          onChangeField={(id, value) => {
            onChangeBlock(block, {
              ...data,
              [id]: value,
            });
          }}
          onChangeBlock={onChangeBlock}
          formData={data}
        />
      </SidebarPortal>
    </React.Fragment>
  );
};

export default compose(injectIntl)(Edit);
