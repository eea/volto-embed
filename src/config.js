import ViewIframeBlock from './Iframe/ViewIframe.jsx';
import EditIframeBlock from './Iframe/EditIframe.jsx';

export function applyConfig(config) {
  config.blocks.blocksConfig.maps = {
    ...config.blocks.blocksConfig.maps,
    view: ViewIframeBlock,
    edit: EditIframeBlock,
  };
  return config;
}
