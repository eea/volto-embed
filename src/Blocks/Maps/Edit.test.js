import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { Provider } from 'react-intl-redux';
import config from '@plone/volto/registry';
import '@testing-library/jest-dom/extend-expect';

import Edit from './Edit';

describe('Test Maps Block editing', () => {
  config.blocks.blocksConfig = {
    ...config.blocks.blocksConfig,
    maps: {
      id: 'maps',
      title: 'Maps',
      group: 'media',
      extensions: {},
      variations: [],
      restricted: false,
      mostUsed: true,
      sidebarTab: 1,
      security: {
        addPermission: [],
        view: [],
      },
    },
  };

  const data = {
    '@type': 'maps',
    url: '<iframe src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d11255.043736345397!2d24.6862147!3d45.15143895!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x474d3248f30cd7b5%3A0x307c5acf21ded9e3!2sHotel%20Posada!5e0!3m2!1sro!2sro!4v1701254795494!5m2!1sro!2sro" width="600" height="450" style="border:0;" allowfullscreen="" loading="lazy" referrerpolicy="no-referrer-when-downgrade"></iframe>',
    dataprotection: {
      enabled: false,
      privacy_statement: 'This is a privacy statement',
    },
    useVisibilitySensor: false,
  };

  it('renders the edit form with a map', () => {
    const { container } = render(
      <Provider store={global.store}>
        <Edit
          data={data}
          pathname="/news"
          selected={false}
          block="1234"
          index={1}
          onChangeBlock={() => {}}
          onSelectBlock={() => {}}
          onDeleteBlock={() => {}}
          onFocusPreviousBlock={() => {}}
          onFocusNextBlock={() => {}}
          handleKeyDown={() => {}}
          content={{}}
        />
      </Provider>,
    );
    expect(screen.getByTitle('ESRI Maps Embedded Block')).toBeInTheDocument();
    expect(container.querySelector('iframe')).toBeInTheDocument();
  });

  it('renders the edit form without a map', () => {
    render(
      <Provider store={global.store}>
        <Edit
          data={{
            '@type': 'maps',
            url: '',
          }}
          pathname="/news"
          selected={true}
          block="1234"
          index={1}
          onChangeBlock={() => {}}
          onSelectBlock={() => {}}
          onDeleteBlock={() => {}}
          onFocusPreviousBlock={() => {}}
          onFocusNextBlock={() => {}}
          handleKeyDown={() => {}}
          content={{}}
        />
      </Provider>,
    );
    expect(
      screen.getByText(
        'Please enter the embed code or URL for the ESRI webmap.',
      ),
    ).toBeInTheDocument();
  });

  it('handles URL input', () => {
    const { container } = render(
      <Provider store={global.store}>
        <Edit
          data={{
            '@type': 'maps',
            url: '',
          }}
          pathname="/news"
          selected={true}
          block="1234"
          index={1}
          onChangeBlock={jest.fn()}
          onSelectBlock={jest.fn()}
          onDeleteBlock={jest.fn()}
          onFocusPreviousBlock={jest.fn()}
          onFocusNextBlock={jest.fn()}
          handleKeyDown={jest.fn()}
        />
      </Provider>,
    );

    const input = screen.getByPlaceholderText('Enter map Embed Code');
    fireEvent.change(input, { target: { value: 'https://example.com/map' } });
    fireEvent.click(container.querySelector('button.primary.button'));

    expect(
      screen.getByDisplayValue('https://example.com/map'),
    ).toBeInTheDocument();
  });
});
