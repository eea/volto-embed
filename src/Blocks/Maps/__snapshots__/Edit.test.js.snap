// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Maps Block editing test-1 1`] = `
<div
  className="block maps align center"
>
  <div
    className="maps-inner"
  >
    <div
      className="privacy-protection-wrapper"
      style={
        Object {
          "height": "auto",
          "minHeight": "200px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <iframe
        allowFullScreen={true}
        className="google-map"
        frameBorder="0"
        src="<iframe src=\\"https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d11255.043736345397!2d24.6862147!3d45.15143895!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x474d3248f30cd7b5%3A0x307c5acf21ded9e3!2sHotel%20Posada!5e0!3m2!1sro!2sro!4v1701254795494!5m2!1sro!2sro\\" width=\\"600\\" height=\\"450\\" style=\\"border:0;\\" allowfullscreen=\\"\\" loading=\\"lazy\\" referrerpolicy=\\"no-referrer-when-downgrade\\"></iframe>"
        style={Object {}}
        title="Google Maps Embedded Block"
      />
    </div>
  </div>
  <div
    className="map-overlay"
  />
  <div
    id="sidebar"
  >
    <div
      className="ui form"
    >
      <header
        className="header pulled"
      >
        <h2>
          Maps
        </h2>
      </header>
      <div
        id="blockform-fieldset-default"
      >
        <div
          className="ui segment form attached"
        >
          <div
            className="mocked-url-widget"
            id="mocked-field-url"
          >
            Maps URL
             - 
            No description
          </div>
          <div
            className="mocked-default-widget"
            id="mocked-field-title"
          >
            Alt text
             - 
            No description
          </div>
          <div
            className="mocked-align-widget"
            id="mocked-field-align"
          >
            Alignment
             - 
            No description
          </div>
          <div
            className="mocked-default-widget"
            id="mocked-field-height"
          >
            <a
              href="https://developer.mozilla.org/en-US/docs/Web/CSS/height"
              rel="noreferrer"
              target="_blank"
            >
              CSS height
            </a>
             - 
            Map height
          </div>
        </div>
      </div>
      <div
        className="accordion ui fluid styled form"
      >
        <div
          id="blockform-fieldset-undefined"
        >
          <div
            className="active title"
            onClick={[Function]}
          >
            Data Protection
            <svg
              className="icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": undefined,
                }
              }
              onClick={null}
              style={
                Object {
                  "fill": "currentColor",
                  "height": "20px",
                  "width": "auto",
                }
              }
              viewBox=""
              xmlns=""
            />
          </div>
          <div
            className="content active"
          >
            <div
              aria-hidden={false}
              className="rah-static rah-static--height-auto"
              style={
                Object {
                  "height": "auto",
                  "overflow": "visible",
                }
              }
            >
              <div
                style={
                  Object {
                    "WebkitTransition": "opacity 500ms ease 0ms",
                    "transition": "opacity 500ms ease 0ms",
                  }
                }
              >
                <div
                  className="ui segment attached"
                >
                  <div
                    className="mocked-default-widget"
                    id="mocked-field-dataprotection"
                  >
                    No title
                     - 
                    No description
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test Maps Block editing test-2 1`] = `
<div
  className="block maps align center"
>
  <div
    className="maps-inner"
  >
    <div
      className="privacy-protection-wrapper"
      style={
        Object {
          "height": "auto",
          "minHeight": "200px",
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <iframe
        allowFullScreen={true}
        className="google-map"
        frameBorder="0"
        src="<iframe src=\\"https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d11255.043736345397!2d24.6862147!3d45.15143895!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x474d3248f30cd7b5%3A0x307c5acf21ded9e3!2sHotel%20Posada!5e0!3m2!1sro!2sro!4v1701254795494!5m2!1sro!2sro\\" width=\\"600\\" height=\\"450\\" style=\\"border:0;\\" allowfullscreen=\\"\\" loading=\\"lazy\\" referrerpolicy=\\"no-referrer-when-downgrade\\"></iframe>"
        style={Object {}}
        title="Google Maps Embedded Block"
      />
    </div>
  </div>
  <div
    id="sidebar"
  >
    <div
      className="ui form"
    >
      <header
        className="header pulled"
      >
        <h2>
          Maps
        </h2>
      </header>
      <div
        id="blockform-fieldset-default"
      >
        <div
          className="ui segment form attached"
        >
          <div
            className="mocked-url-widget"
            id="mocked-field-url"
          >
            Maps URL
             - 
            No description
          </div>
          <div
            className="mocked-default-widget"
            id="mocked-field-title"
          >
            Alt text
             - 
            No description
          </div>
          <div
            className="mocked-align-widget"
            id="mocked-field-align"
          >
            Alignment
             - 
            No description
          </div>
          <div
            className="mocked-default-widget"
            id="mocked-field-height"
          >
            <a
              href="https://developer.mozilla.org/en-US/docs/Web/CSS/height"
              rel="noreferrer"
              target="_blank"
            >
              CSS height
            </a>
             - 
            Map height
          </div>
        </div>
      </div>
      <div
        className="accordion ui fluid styled form"
      >
        <div
          id="blockform-fieldset-undefined"
        >
          <div
            className="active title"
            onClick={[Function]}
          >
            Data Protection
            <svg
              className="icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": undefined,
                }
              }
              onClick={null}
              style={
                Object {
                  "fill": "currentColor",
                  "height": "20px",
                  "width": "auto",
                }
              }
              viewBox=""
              xmlns=""
            />
          </div>
          <div
            className="content active"
          >
            <div
              aria-hidden={false}
              className="rah-static rah-static--height-auto"
              style={
                Object {
                  "height": "auto",
                  "overflow": "visible",
                }
              }
            >
              <div
                style={
                  Object {
                    "WebkitTransition": "opacity 500ms ease 0ms",
                    "transition": "opacity 500ms ease 0ms",
                  }
                }
              >
                <div
                  className="ui segment attached"
                >
                  <div
                    className="mocked-default-widget"
                    id="mocked-field-dataprotection"
                  >
                    No title
                     - 
                    No description
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
